Feature/commands beep leer store scene

maximilianriemensberger opened the pull request:
Issues #15 #22 

Add further commands to `PckGenerator`:
* `beep` makes a module beep up to 15 times
* `do_nothing` does nothing and can used to discover and ping modules
* `store_scene_*` commands can store the current state of outputs and relays in scene-registers

@alengwenus Please review.  As far as I understand the random and  incomplete pieces of information that I could find, the scene commands for activation and store are identical except for the action flags `"S"` and `"A"`. Similarly for the beep command.

alengwenus commented on pypck/pck_commands.py:
```diff
@@ -991,11 +1029,66 @@ def activate_scene_relay(
+        """
+        if (count < 1) or (count > 15):
+            raise ValueError("Wrong number of beeps.")
+        return "PI{:s}{:02d}".format(sound, count)
```
Not necessarily wrong, but for decimal numbers we should pad to three digits unless otherwise expected by PCK syntax (`{:03d}` for count format).

alengwenus commented on pypck/pck_commands.py:
```diff
@@ -991,11 +1029,66 @@ def activate_scene_relay(
+    def do_nothing() -> str:
+        """Generate a command that does nothing.
+
+        Combine with request for acknoledgement to discover and
```
typo:  acknowledgement

alengwenus reviewed (changes requested):
We should also think about adding corresponding convenience methods to `module.AbstractConnection` (see e,g, `AbstractConnection.dim_output`).
Especially when sending commands from within HomeAssistant, this is desirable.

maximilianriemensberger commented:
@alengwenus I addressed your comments. There are just two things I was wondering about:

* Is there a better name for the LEER command instead of `do_nothing`.  In AbstractConnection it feels a tiny bit confusing as one might not expect that it sends a command. Suggestions for a better name are welcome.
* Since the ack flag is set to `not self.is_group()`, I'm not even sure if the `do_nothing()` convenience function is very useful in AbstractConnection. But probably it is, at least for modules.

An one more question: As far as I can tell pypck doen't have functionality to discover group memberships. Would something like this be desirable? I think it could be at least useful for showing the membership as metadata or details somewhere.

alengwenus commented:
> * Is there a better name for the LEER command instead of `do_nothing`.  In AbstractConnection it feels a tiny bit confusing as one might not expect that it sends a command. Suggestions for a better name are welcome.

In my private developments and docs I usually call it the `Empty` command. ;-)

> * Since the ack flag is set to `not self.is_group()`, I'm not even sure if the `do_nothing()` convenience function is very useful in AbstractConnection. But probably it is, at least for modules.

My suggestion would be to replace the `not self.is_group()` statement with `True` and simply call the convenience function `ping`?

> An one more question: As far as I can tell pypck doen't have functionality to discover group memberships. Would something like this be desirable? I think it could be at least useful for showing the membership as metadata or details somewhere.

I think this should be possible. I opened an issue for that: #27

maximilianriemensberger commented:
I implemented your suggestions regarding the LEER command.

alengwenus reviewed (approved):
Great! Thanks a lot @maximilianriemensberger! 👍
